### PR TITLE
Birthstone: Convert to full template

### PR DESCRIPTION
**Pull request by rasikapohankar**
Updates BirthStone Goodie Re:  #2767.
IA Page: https://duck.co/ia/view/birth_stone

Maintainer: @austinheimark

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by mintsoft** [comment]
Addressing: https://github.com/duckduckgo/zeroclickinfo-goodies/pull/2792#issuecomment-201914195

File: share/goodie/cheat_sheets/json/language/spanish.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
             {
                 "key": "Please",
                 "val": "Por favor.",
-		        "trn": "pohr fah-BOHR"
+                "trn": "pohr fah-BOHR"
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/language/spanish.json]: can you remove this change from this PR?

**Comment by rasikapohankar** [comment]
Hey @mintsoft! Removed spanish.json from the commit. Do you still see it?

**Comment by mintsoft** [comment]
@rasikapohankar Yeah; if you look at: https://github.com/duckduckgo/zeroclickinfo-goodies/pull/2807/files you can see it's in the diff

**Comment by mintsoft** [comment]
@austinheimark I'm not sure why you've not been cc'd on this automatically; however you are the maintainer of BirthStone so any input you'd like to offer would be much appreciated.

**Comment by rasikapohankar** [comment]
@mintsoft : I changed the value of 'showing changes from all commits' to  'Removed spanish.json from commit.', then it shows two files. Is that right?

**Comment by mintsoft** [comment]
@rasikapohankar that changes the view; however the actual PR will still contain the changes that were made to `spanish.json`; you need to "undo" all the changes made to that file in order for it to not appear on the diff anymore

**Comment by rasikapohankar** [comment]
@mintsoft : I haven't made changes to spanish.json recently. spanish.json and countdown.js both are from a different branches. I did not understand why they got included here. To undo it, I did : 1. git reset --soft HEAD^, 2. git reset file. And then did commit and push again. But it is still showing. What should I do?

**Comment by mintsoft** [comment]
@rasikapohankar the easiest option is to clone master somewhere else; copy the `spanish.json` over the top of the copy that's in your branch and commit it in. If the files are identical; it won't show up on this diff.

**Comment by rasikapohankar** [comment]
@mintsoft: I did something wrong and now it cannot find the branches I had before. Should I re-do the changes and open a pull request again? I have copied the changes.

**Comment by mintsoft** [comment]
@rasikapohankar you can do that if you're more comfortable that way. However the branch is called `birthstone`; you should be able to find it with a fresh checkout:
```
git clone https://github.com/rasikapohankar/zeroclickinfo-goodies.git;
git checkout birthstone
```

**Comment by rasikapohankar** [comment]
@mintsoft: Deleted the file spanish.json and did a commit. Now there are only the needed two files?

**Comment by mintsoft** [comment]
@rasikapohankar The problem is now that if I merge this PR; it will delete spanish.json. The file needs to exist in your branch but be identical to the one in master

**Comment by rasikapohankar** [comment]
@mintsoft: I copied the file contents from the one merged to this file now.

**Comment by rasikapohankar** [comment]
@mintsoft : It's showing files changed as two now instead of three.

**Comment by mintsoft** [comment]
@rasikapohankar that looks good now; there's only 2 files in the diff; once we hear from the maintainer we can move onward :+1:

**Comment by rasikapohankar** [comment]
@mintsoft : Okay! 
I must have changed the file after it got merged and that's why it was showing here? I think I had changed the indentation but did not push the changes.

File: lib/DDG/Goodie/BirthStone.pm [hunk 2]
```diff
@@ -30,13 +30,19 @@ handle remainder => sub {
...
     return unless $stone;
-
-    return $month . " birthstone: $stone",
-      structured_answer => {
-        input     => [$month],
-        operation => 'Birthstone',
-        result    => $stone
-      };
+    
+    return "",
```
Comment (moollaza) [inline lib/DDG/Goodie/BirthStone.pm]: we should still return a plaintext result when possible. In this case we can, so we should.

**Comment by moollaza** [comment]
@rasikapohankar thanks for this!

@mintsoft I don't feel we need maintainer approval for these. This is a necessary change and it doesn't affect the content of the IA, just the way it's being displayed.

File: lib/DDG/Goodie/BirthStone.pm (hunk 2 above)
Comment (rasikapohankar) [inline lib/DDG/Goodie/BirthStone.pm]: @moollaza : Yes, I was just going to ask. What is the significance of that return value before structured_answer?

File: lib/DDG/Goodie/BirthStone.pm [hunk 3]
```diff
@@ -30,13 +30,19 @@ handle remainder => sub {
...
-        input     => [$month],
-        operation => 'Birthstone',
-        result    => $stone
-      };
+    structured_answer => {
+            id => 'birthstone',
+            name => 'Answer',
+            data => {
+                title => $stone,
+                subtitle => 'Birthstone: '.$month
```
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/BirthStone.pm]: Maybe `"Birthstone for $month'`.

File: t/BirthStone.t [hunk 4]
```diff
@@ -8,48 +8,33 @@ use DDG::Test::Goodie;
...
-            input     => ['May'],
-            operation => 'Birthstone',
-            result    => 'Emerald'
-        }
-    ),
+    'april birth stone' => build_test('April', 'Diamond'),
+    'birthstone JUNE' => build_test('June', 'Pearl'),
+    'DecEmber birthstone' => build_test('December', 'Turquoise'),
+    'birthstone april' => build_test('April', 'Diamond'),
+    'may birth stone' => build_test('May', 'Emerald')
```
Comment (GuiltyDolphin) [inline t/BirthStone.t]: Does `:Tabularize /=>` mean anything to you?



If not, you're missing out. But lets align these :wink:

**Comment by GuiltyDolphin** [comment]
@rasikapohankar I notice you've been having some commit issues - for future reference it might be worth re-forking so you can get a fully-clean master to branch off of in the future.

File: lib/DDG/Goodie/BirthStone.pm (hunk 3 above)
Comment (rasikapohankar) [inline lib/DDG/Goodie/BirthStone.pm]: @GuiltyDolphin : It was Birthstone: month in the original instant answer. Should I change it?

**Comment by rasikapohankar** [comment]
@GuiltyDolphin : Thanks! I will do that the next time before making a new branch. Sorry I did not know Tabularize. Read about it now :smiley:. 
The subtitle was Birthstone: month in the original instant answer. Should I change it?

File: lib/DDG/Goodie/BirthStone.pm (hunk 3 above)
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/BirthStone.pm]: @rasikapohankar Hmm... You can leave it as is and we can get it sorted later on, or if you want to go ahead and change it we can ping the maintainer and make sure they're happy with it.

**Comment by rasikapohankar** [comment]
@GuiltyDolphin : I have changed the subtitle to Birthstone for (month).

**Comment by GuiltyDolphin** [comment]
@rasikapohankar Thanks :dolphin:

**Pull request merged**